core/src/main/scala/org/http4s/rho/AuthedServiceBuilder.scala

```diff
@@ -0,0 +1,39 @@
+package org.http4s.rho
+
+import org.http4s._
+import org.http4s.rho.CompileService.foldServices
+import org.http4s.rho.bits.PathTree
+import org.log4s._
+import shapeless.HList
+
+import scala.collection.immutable.VectorBuilder
+
+/** CompileService which accumulates routes and can build a `AuthedService` */
+final class AuthedServiceBuilder[U] private(internalRoutes: VectorBuilder[RhoRoute.Tpe]) extends ServiceBuilder(internalRoutes) {
```

Comment:
I don't think we need this class at all: the important part
```scala
def toAuthedService(filter: RhoMiddleware = identity): AuthedService[U] = {
  val rhoService = toService(filter)
  Service.lift { case AuthedRequest(authInfo, req) =>
    rhoService(req.withAttribute[U](authKey, authInfo))
  }
}
```
can be inlined into `AuthedRhoService`.